pandas/core/frame.py

```diff
@@ -2909,7 +2909,7 @@ def transpose(self, *args, copy: bool = False) -> DataFrame:
             arr_type = dtype.construct_array_type()
             values = self.values
 
-            new_values = [arr_type._from_sequence(row, dtype=dtype) for row in values]
+            new_values = [arr_type._from_scalars(row, dtype=dtype) for row in values]
```

Comment:
in this case we already know we have the correct types, so cant we go directly to from_sequence?